Doc/tools/extensions/pyspecific.py

```diff
@@ -254,11 +254,8 @@ def run(self):
         fpath = path.join(source_dir, fname)
         self.state.document.settings.record_dependencies.add(fpath)
         try:
-            fp = codecs.open(fpath, encoding='utf-8')
-            try:
+            with io.open(fpath, encoding='utf-8') as fp:
```

Comment:
Maybe some people or services still use Python 2 to build the Python documentation, no?